File: News_Service_WebFlux/src/main/java/com/api/AdvancedServiceWebFlux/dto/common/CustomSuccessResponse.java
package com.api.AdvancedServiceWebFlux.dto.common;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CustomSuccessResponse<T> extends BaseSuccessResponse {

    private final T data;

    public CustomSuccessResponse(T data) {
        this.data = data;
    }

    public CustomSuccessResponse(int statusCode, T data) {
        super(statusCode);
        this.data = data;
    }
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/dto/common/CustomSuccessResponse.kt
package com.dunice.advancedServiceKotlin.dto.common

class CustomSuccessResponse<T> : BaseSuccessResponse {
    val data: T

    constructor(data: T) {
        this.data = data
    }

    constructor(statusCode: Int, data: T) : super(statusCode) {
        this.data = data
    }
}
